### DISCUSSION TITLE
scalar kernel for average of coupled nonlinear variable value

### DISCUSSION THREAD

**Discussion by yuegu2013** [idaholab/moose#17098] [Q&A General] [answered]
Hello,

Is there an existing Scalar kernel that assigns the average (element or nodal) of a coupled nonlinear variable to it? 

This can be achieved by adding a postprocessor and then assign the postprocessor value to the scalar; however, this misses the off-diagonal jacobian term. 

This seems to be a common need. I looked through the MOOSE source code, but did not find one. Can you please help point me to the right class, if there is one? 

Thanks,
Guojun

**Comment by GiudGiud** [comment, marked as answer]
Hi

I dont think so but we would welcome a PR.

Guillaume
